src/stackwalk.c

```diff
@@ -697,6 +697,161 @@ void jl_print_bt_entry_codeloc(jl_bt_element_t *bt_entry) JL_NOTSAFEPOINT
     }
 }
 
+
+#ifdef _OS_LINUX_
+#if defined(__GLIBC__) && defined(_CPU_AARCH64_)
+#define LONG_JMP_SP_ENV_SLOT 13
+static uintptr_t julia_longjmp_xor_key;
+// GLIBC mangles the function pointers in jmp_buf (used in {set,long}*jmp
+// functions) by XORing them with a random key.  For AArch64 it is a global
+// variable rather than a TCB one (as for x86_64/powerpc).  We obtain the key by
+// issuing a setjmp and XORing the SP pointer values to derive the key.
+static void JuliaInitializeLongjmpXorKey(void)
+{
+    // 1. Call REAL(setjmp), which stores the mangled SP in env.
+    jmp_buf env;
+    _setjmp(env);
+
+    // 2. Retrieve vanilla/mangled SP.
+    uintptr_t sp;
+    asm("mov  %0, sp" : "=r" (sp));
+    uintptr_t mangled_sp = ((uintptr_t*)&env)[LONG_JMP_SP_ENV_SLOT];
+
+    // 3. xor SPs to obtain key.
+    julia_longjmp_xor_key = mangled_sp ^ sp;
+}
+#endif
+
+JL_UNUSED static uintptr_t ptr_demangle(uintptr_t p)
+{
+#if defined(__GLIBC__)
+#if defined(_CPU_X86_)
+// from https://github.com/bminor/glibc/blame/master/sysdeps/unix/sysv/linux/i386/sysdep.h
```

Comment:
It'd be nicer to use permalinks instead of referencing master

Reply:
these intentionally reference blame, since if the code changes any, I want that to be very obvious there to the reader